tls/s2n_psk.h

```diff
@@ -33,6 +33,18 @@ typedef enum {
     S2N_PSK_DHE_KE,
 } s2n_psk_key_exchange_mode;
 
+struct s2n_psk_identity{
+    uint8_t *identity;
+    size_t identity_length;
+};
+
+struct s2n_pre_shared_key {
+    struct s2n_psk_identity identity;
+    uint8_t *secret;
+    size_t secret_len;
+    s2n_hmac_algorithm hmac;
```

Comment:
We talked offline, but you'll want to use an external enum, probably without all the options that can't be used in TLS1.3 / for the KDF.